srcjs/bindings/messagesInputBinding.js

```diff
@@ -65,42 +66,39 @@ $.extend(f7MessagesBinding, {
 
   // Given the DOM element for the input, return the value
   getValue: function(el) {
-    var messages = this.app.messages.get(el).messages;
     // Ignore the first message as it is set by the template
-    messages = messages.slice(1);
+    var m = this.messages.slice(1);
     // Return an object of objects: will be a list of lists in R
-    return messages.length > 0 ? { ...messages } : null;
+    return m.length > 0 ? { ...m } : null;
   },
 
   // see updatef7Messages
   setValue: function(el, value) {
-    var messages = this.app.messages.get(el);
+    var self = this;
     var message = value.value;
     // TO DO: does not work if TRUE
     if (value.showTyping) {
```

Comment:
@hypebright I could not make this one working. You can try with this below example and put a `console.log(data)` in the `receiveMessage` messages binding method. You'll see that after sending your message through the app, the second message is empty which does not make much sense since updateF7Messages sends the correct JSON. Something with `session$sendInputMessages()`? Something JS side? Maybe we don't need a binding after all and just use `session$sendCustomMessage` + some JS code like for other widgets (gauges...). I'll continue working on the other elements.

```r
shinyApp(
  ui = f7Page(
    title = "Messages",
    f7SingleLayout(
      navbar = f7Navbar(
        title = "Messages",
        hairline = FALSE,
        shadow = TRUE
      ),
      toolbar = f7MessageBar(inputId = "mymessagebar", placeholder = "Message"),
      # main content
      f7Messages(id = "mymessages", title = "My message")

    )
  ),
  server = function(input, output, session) {
    messageSent <- reactiveVal()
    observeEvent(input[["mymessagebar-send"]], {
      messageSent(FALSE)
      updateF7Messages(
        id = "mymessages",
        list(
         f7Message(
          text = input$mymessagebar,
          name = "David",
          type = "sent",
          header = "Message Header",
          footer = "Message Footer",
          textHeader = "Text Header",
          textFooter = "text Footer",
          avatar = "https://cdn.framework7.io/placeholder/people-100x100-7.jpg"
         )
        )
      )
      messageSent(TRUE)
    })

    observeEvent(req(messageSent()), {
      names <- c("Victor", "John")
      name <- sample(names, 1)

      updateF7Messages(
        id = "mymessages",
        showTyping = TRUE,
        list(
         f7Message(
          text = "Some message",
          name = name,
          type = "received",
          avatar = "https://cdn.framework7.io/placeholder/people-100x100-9.jpg"
         )
        )
      )
    })

  }
 )
```